### ISSUE TITLE
Move the methods related to Ruby from the Asciidoctor API to internals

### ISSUE THREAD

**Issue by Mogztter** [asciidoctor/asciidoctorj#714]
Specifically:

* [ ] `void requireLibrary(String... requiredLibraries)`
* [ ] `void requireLibraries(Collection<String> requiredLibraries)`
* [ ] `RubyExtensionRegistry rubyExtensionRegistry()`

I'm not really sure how we should handle this case.
If a user want to require a (Ruby) library and/or to get the `RubyExtensionRegistry` they can cast the `Asciidoctor` interface to `JRubyAsciidoctor` and call the methods ?
But that mean that they would have to use the `asciidoctorj-core` module as a "compile" dependency and not a runtime dependency.

Is there a benefit to move all the JRuby related code out of the `asciidoctorj-core` into a `asciidoctorj-jruby` ?

**Comment by mojavelinux** [comment]
This is where the pattern `getDelegate()` might apply well (like in JPA). We want to step from the generic interface to the vendor-specific interface to access additional implementation-specific methods.

```
((JRubyAsciidoctor) asciidoctor.getDelegate()).requireLibrary('asciimath')
```

We could also consider an API where the expected class is passed in:

```
asciidoctor.getDelegate(JRubyAsciidoctor.class).requireLibrary('asciimath')
```

If anyone is aware of similar competing patterns, please bring them forward.

**Comment by robertpanzer** [comment]
> `asciidoctor.getDelegate(JRubyAsciidoctor.class).requireLibrary('asciimath')`

I like this. 
Or maybe `.unwrap()` instead of `.getDelegate()`? 
```
asciidoctor.unwrap(JRubyAsciidoctor.class).requireLibrary('asciimath')
```
I wonder though if we should make this change with 2.0 and keep 1.6 compatible with the current build plugins.
In particular as we already have the RC out.

**Comment by mojavelinux** [comment]
Yes, this feels 2.0 to me.

**Comment by Mogztter** [comment]
Yep, if you've already released a RC then this change shouldn't be part of the final release.
AsciidoctorJ 1.6.0 is a "major" AsciidoctorJ but still based on Asciidoctor (Ruby) 1.5.8 ?
